corehq/blobs/zipdb.py

```diff
@@ -38,8 +38,17 @@ def get_path(self, identifier=None, bucket=DEFAULT_BUCKET):
 
     def exists(self, identifier, bucket=DEFAULT_BUCKET):
         path = self.get_path(identifier, bucket)
-        with zipfile.ZipFile(self.zipname, 'r') as z:
-            return path in z.namelist()
+        return path in self.zipfile.namelist()
+
+    @property
+    def zipfile(self):
+        if self._zipfile is None:
+            self._zipfile = zipfile.ZipFile(self.zipname, 'w', allowZip64=True)
+        return self._zipfile
+
+    def close(self):
+        if self._zipfile:
+            self._zipfile.close()
```

Comment:
Hmm. yeah. I didn't realize that it would take that long 24 h to 10 min seems unreal.